TEMPLATES/mega-linter.yml

```diff
@@ -40,129 +120,231 @@ jobs:
+      # https://github.com/actions/checkout
+      #
       - name: Checkout Code
         uses: actions/checkout@v4
         with:
           token: ${{ secrets.PAT || secrets.GITHUB_TOKEN }}
 
-          # If you use VALIDATE_ALL_CODEBASE = true, you can remove this line to
-          # improve performance
+          ##
+          # A fetch depth of `0` pulls the entire Repository's history,
+          # Branches, and Tags. Limiting this to `1` (or commenting out the
+          # line, as this is the default) increases efficiency, but is only
+          # applicable when `VALIDATE_ALL_CODE_BASE = true`.
+          #
           fetch-depth: 0
 
-      # MegaLinter
+      ##
+      # Run MegaLinter.
+      #
+      # More info at:
+      # https://github.com/oxsecurity/megalinter
+      #
       - name: MegaLinter
+        id: ml
 
-        # You can override MegaLinter flavor used to have faster performances
-        # More info at https://megalinter.io/flavors/
+        ##
+        # The full MegaLinter project can be over 8GB in size! This is due to it
+        # including every linter available for the full MegaLinter project. It
+        # is highly recommended to select a more refined "flavor" that better
+        # matches your project by changing the image used here.
+        #
+        # For example, a Python project may elect to use:
+        #
+        #   oxsecurity/megalinter/flavors/python@v7
+        #
+        # By default, MegaLinter recommends a specific flavor for your project
+        # setup in its logs. It's shown after the summary table at the end of
+        # the run for your convenience.
+        #
+        # More info at:
+        # https://megalinter.io/flavors/
+        #
         uses: oxsecurity/megalinter@v7
 
-        id: ml
-
-        # All available variables are described in documentation
-        # https://megalinter.io/configuration/
+        ##
+        # These variables will override project configurations, including those
+        # set in the `.mega-linter.yml` file. This is useful for creating
+        # conditional exceptions during GitHub workflows, for example, when
+        # linting a production deployment.
+        #
+        # https://megalinter.io/latest/config-file/
+        #
         env:
 
-          # Validates all source when push on main, else just the git diff with
-          # main. Override with true if you always want to lint all sources
+          ##
+          # By default, this validates full codebase whenever a push to the
+          # default branch occurs. Otherwise, only the `git diff` files will
+          # be linted for better efficiency.
           #
-          # To validate the entire codebase, set to:
-          # VALIDATE_ALL_CODEBASE: true
+          # This can be overridden to always lint the codebase with:
+          # 
+          #   VALIDATE_ALL_CODEBASE: true
+          # 
+          # To validate only `git diff`` with the default branch on push:
```

Comment:
```suggestion
          # This can be overridden to always lint the codebase with:
          #
          #   VALIDATE_ALL_CODEBASE: true
          #
          # To validate only `git diff`` with the default branch on push:
```